docs/concepts/functional-components/macro.md

```diff
@@ -0,0 +1,33 @@
+---
+title: #[functional_component]
+description: The #[functional_component] macro
+---
+
+
+The `#[functional_component]` macro allows you automatically to generate the requirements for creating Yew functional components. Any Rust function can be annotated and used as a functional component. 
+
+A function annotated with `#[functional_component]` may take one argument for props of `Properties` type's reference and must return `Html`. The name of component is passed as an attribute to the said macro. Note that unlike struct component, the `Properties` type for functional components must also implement `PartialEq`.
+
+### Example
+
+```rust
+#[derive(Clone, Properties, PartialEq)]
+struct Props {
+    a: usize,
+}
+
+#[functional_component(Comp)]
+fn comp(props: &Props) -> Html {
+    html! {
+        <p>
+            { props.text }
+        </p>
+    }
+}
```

Comment:
How about we use a slightly simplified version of the original "counter" example you had here?

Reply:
The counter example would require the use of hooks. Is that something that should be used in documentation without any mentioning of what they are/do?

Reply:
Of course. It's not ideal that hooks aren't documented yet, of course, but all of this isn't supposed to be used yet anyway (it might even be merged into the yew crate before being released).

Reply:
Okay, I might add *some* documentation with PR. This changed example will be a little big though.

Reply:
Huh, I didn't want to give the impression that you had to add documentation for hooks. If you want to then sure go ahead (though maybe consider doing it in a separate PR to keep this focused).

> This changed example will be a little big though.

That's why I said slightly simplified. But the original example wasn't that long so it shouldn't be that much of a problem.

Reply:
I have already added some placeholders for those. The documentation for those is something to be done on a later day though.